src/cmds/restic/cmd_find.go

```diff
@@ -192,33 +178,14 @@ func runFind(opts FindOptions, gopts GlobalOptions, args []string) error {
 		}
 	}
 
-	err = repo.LoadIndex()
-	if err != nil {
+	if err = repo.LoadIndex(); err != nil {
 		return err
 	}
 
-	pat.pattern = args[0]
-
-	if opts.CaseInsensitive {
-		pat.pattern = strings.ToLower(pat.pattern)
-		pat.ignoreCase = true
-	}
-
-	if opts.Snapshot != "" {
-		snapshotID, err := restic.FindSnapshot(repo, opts.Snapshot)
-		if err != nil {
-			return errors.Fatalf("invalid id %q: %v", args[1], err)
-		}
-
-		return findInSnapshot(repo, pat, snapshotID)
-	}
-
-	done := make(chan struct{})
-	defer close(done)
-	for snapshotID := range repo.List(restic.SnapshotFile, done) {
-		err := findInSnapshot(repo, pat, snapshotID)
-
-		if err != nil {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Uhm. This is `main()` or equivalent. Unless cobra is going to supply us with a context, this is as high in the tree as we can get. So `context.Background()` felt like the right choice there.

For me, `context.TODO()` would appear in say refactoring the backends, when the restic library itself does not plumb the context through, so you would have to make one up.